Add omitted self.

sbellem opened the pull request:
Please forgive if I am the one omitting a detail as it appeared to me that there is no variable `round`, but `self.round` there is.

[`round`](https://docs.python.org/2/library/functions.html#round) being a built-in in Python the expression:

```python
assert r >= round
```

should throw an `AssertionError`, e.g.:

```python
>>> assert 3 >= round
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```

Since that part of the code is [untested](https://codecov.io/gh/sbellem/honeybadgerbft/src/travis-ci/honeybadgerbft/core/honeybadger.py#L43), a test should be added as part of this PR.

- [x] Test https://github.com/amiller/HoneyBadgerBFT/blob/dev/honeybadgerbft/core/honeybadger.py#L41-L44 _at least_

sbellem committed:
Add omitted `self.`

amiller commented:
Agreed that this is a buggy line of code, and also that this issue should require us to show test coverage for that codepath too.

It looks like this line of code should be run in any schedule where some node receives a message for the next block `r+1` before determining that block `r` has been committed. But that should happen easily under ordinary circumstances, so I'm surprised this isn't triggered by the multi-block test. https://github.com/amiller/HoneyBadgerBFT/blob/dev/test/test_honeybadger.py

sbellem commented:
more of a note to myself, but just posting here the result of a run that did go through the if clause ...

(seems to be random)

```bash
root@70bb7f2a3246:/usr/local/src/HoneyBadgerBFT# pytest -v test/test_honeybadger.py -s
Test session starts (platform: linux2, Python 2.7.12, pytest 3.2.1, pytest-sugar 0.9.0)
cachedir: .cache
rootdir: /usr/local/src/HoneyBadgerBFT, inifile:
plugins: hypothesis-3.21.3, cov-2.5.1, xdist-1.20.0, sugar-0.9.0, forked-0.2, env-0.6.2
####################################################################################################

HYPOTHESIS STRATEGY PARAMS: (N, f) = (4, 1)

####################################################################################################

submit_tx 0 <[HBBFT Input 0]>
submit_tx 1 <[HBBFT Input 1]>
submit_tx 2 <[HBBFT Input 2]>
submit_tx 3 <[HBBFT Input 3]>
submit_tx 0 <[HBBFT Input 10]>
submit_tx 1 <[HBBFT Input 11]>
submit_tx 2 <[HBBFT Input 12]>
submit_tx 3 <[HBBFT Input 13]>
submit_tx 0 <[HBBFT Input 20]>
submit_tx 1 <[HBBFT Input 21]>
submit_tx 2 <[HBBFT Input 22]>
submit_tx 3 <[HBBFT Input 23]>
0 0 tx_to_send: <[HBBFT Input 0]>
1 0 tx_to_send: <[HBBFT Input 1]>
2 0 tx_to_send: <[HBBFT Input 2]>
3 0 tx_to_send: <[HBBFT Input 3]>
new_tx: ('<[HBBFT Input 0]>', '<[HBBFT Input 1]>', '<[HBBFT Input 2]>', '<[HBBFT Input 3]>')
3 1 tx_to_send: <[HBBFT Input 13]>
> /usr/local/src/HoneyBadgerBFT/honeybadgerbft/core/honeybadger.py(44)_recv()
     43                     import ipdb; ipdb.set_trace()  # XXX BREAKPOINT
---> 44                     assert r >= round
     45                     self._per_round_recv[r] = Queue()

ipdb> pp r
1
ipdb> pp self._per_round_recv
{0: <Queue at 0x7fb405d393d0 queue=deque([(2, ('TPKE', 0, [[5438088469139462206047492728488387247960092196552276132625516534562905316561235811446673218794526773430819463810048084511828028689823876913532362066550971, 3280974891085610162233771187430545890930019127394883820350732024213405212879926169744789491669229395368278002835520876167731655496713011093457271981380021], [3073884045985439536743048130889413729544724422811130467342362438235238833165617756737944460921258561459699940611584624341607036339984206409635507448168068, 2167330557959955292415713275083539249218635757548472041422188624492223082746838808625095941601988708253229377956053815236520212634531943147848958813842872], [2098145091870620697976671377080059930826415772285832808180379483473465541375672726848047241670660096908382613685753953800628829776034888272319962690579354, 1423496130246757495440383271530948985117663485051040720113494167582246562911259939849859780758690290455544551788142083266038933265601585699928017548851516], [7565872156391186937584131079548338294685787928595688105598591434438067331604179565071870627473967272350483859386236355323981732425507962065136023929557343, 8517241852444759373058867292566987659059579495243392537743644993752302084933582110439771348865134610357247327728927190217325498513914723161739077690467850]])), (3, ('ACS_ABA', 2, ('EST', 3, 1))), (0, ('ACS_ABA', 2, ('EST', 3, 1))), (0, ('ACS_ABA', 0, ('AUX', 3, 1))), (3, ('ACS_ABA', 0, ('AUX', 3, 1))), (1, ('ACS_ABA', 0, ('AUX', 3, 1))), (3, ('TPKE', 0, [[2026924660162834341640688314116202774593515660231146265963557750395134624503806402931015439536452731138398862951292661563812151879254972508902829691097407, 8549113061091411550593218114394440711185289376776647136742804101691594873111524603297383325987423877949276147162474441858909918160453788717957049921145224], [4869641182903591360510910466977629990676583603357974849988571622271709147286523947502704911402499328397374149125886640932604540611671344776172237400002497, 1670649328932313408313706870663003135983041320178040944357393836320382566953098435751270861809776300445922211092435508773020734979862844906282562595573037], [6958614696888198925983570460315107619373151858759298294061933604191844749996226095940877204259728585294276606877769562376740873110089285870399482549801497, 2660065012790526458392475044690582675470656838319921528044128544935564173408103686452850113927935402799731154134548408312773126006691791676258101534927459], [522674583126403967526561038781883242513736849302139184815643565683218273872515995218720392895626352390549727618980949490493934408008871701956151346721430, 6889882414170203633219542200805467282626184470924087936136155808604902844894492489941326258386427770077353910433636206791106525042673620525777358844773101]]))]) getters[1]>}
ipdb> pp sender
3
```

sbellem committed:
Increase asynchronicity in test_honeybadger

this helps in testing the case in which a new round (r+1) is kicked off
for some nodes meanwhile at least one other node has not yet completed
the previous round (r)

sbellem committed:
Skip code coverage on assert

sbellem committed:
Set concurrency for coverage

amiller reviewed (approved):
This looks good, improves the test realism.
